qa/L0_server_launch_modes/test.sh

```diff
@@ -109,41 +102,103 @@ function run_server_launch_modes() {
 
         # Run the analyzer and check the results, enough to just profile the server
         set +e
-        MODEL_ANALYZER_SUBCOMMAND="profile"
-        run_analyzer
-        if [ $? -ne 0 ]; then
-            echo -e "\n***\n*** Test with launch mode '${LAUNCH_MODE}' using ${PROTOCOL} client Failed."\
-                    "\n***     model-analyzer exited with non-zero exit code. \n***"
+        _run_analyzer
+        _check_gpus $@
+        set -e
+
+        rm -rf $OUTPUT_MODEL_REPOSITORY
+    done
+}
+
+function run_model_modes() {
+    if [ "$LAUNCH_MODE" != "remote" ]; then 
+        echo -e "\n***\n*** Test Failed. Function 'run_model_modes' can only be run in remote mode (not '${LAUNCH_MODE}' mode). \n***"
+        exit 1
+    fi
+
+    model_mode_combos=('--model-control-mode=explicit,' ',--reload-model-disable')
+
+    for model_mode_combo in ${model_mode_combos[@]}
+    do
+        # delete previous array/list (this is crucial!)
+        unset model_mode
+
+        # make array from simple string
+        model_mode=(${model_mode_combo//,/ })
+        MODEL_CONTROL_MODE=${model_mode[0]}
+        RELOAD_MODEL_DISABLE=${model_mode[1]}
+
+        set +e
+        _run_server $@
+        _run_analyzer
+        _kill_server
+        _check_gpus $@
+        set -e
+    done
+}
+
+function _run_server() {
+    if [ "$LAUNCH_MODE" != "remote" ]; then 
+        echo -e "\n***\n*** Test Failed. Function '_run_server' can only be run in remote mode (not '${LAUNCH_MODE}' mode). \n***"
+        exit 1
+    fi
+
+    # For remote launch, set server args and start server
+    SERVER=`which tritonserver`
+    SERVER_ARGS="--model-repository=$MODEL_REPOSITORY $MODEL_CONTROL_MODE --http-port $http_port --grpc-port $grpc_port --metrics-port $metrics_port"
```

Comment:
Notice how `--model-control-mode=explicit` is replaced with `$MODEL_CONTROL_MODE`